library/core/src/num/nonzero.rs

```diff
@@ -2265,3 +2265,54 @@ mod verify {
     nonzero_check!(u128, core::num::NonZeroU128, nonzero_check_new_unchecked_for_u128);
     nonzero_check!(usize, core::num::NonZeroUsize, nonzero_check_new_unchecked_for_usize);
 }
+
+#[cfg(kani)]
+mod macro_nonzero_check_clamp {
+    use super::*;
+    macro_rules! nonzero_check_clamp {
+        ($t:ty, $nonzero_type:ty, $nonzero_check_clamp_for:ident) => {
+            #[kani::proof]
+            pub fn $nonzero_check_clamp_for() {
+                let x: $t = kani::any();
+                kani::assume(x != 0);
+                let min: $t = kani::any();
+                kani::assume(min != 0);
+                let max: $t = kani::any();
+                kani::assume(max != 0);
+                
+                // Ensure min <= max
+                kani::assume(min <= max);
```

Comment:
`clamp` is a safe function, so there shouldn't be a need for a such a precondition?!

Reply:
`clamp` will panic if `min > max`. @lang280 Can you write another macro that leverages Kani's `should_panic` feature to check for this case? [Here](https://github.com/model-checking/verify-rust-std/blob/d0d9de25c44d829f52c837636284cbfa776f544e/library/core/src/ptr/non_null.rs#L1910) is an example.